src/NUnitTestAdapter/Internal/AsyncMethodHelper.cs

```diff
@@ -24,7 +24,7 @@ public string GetClassNameForAsyncMethod(string className, string methodName)
             var definingType = targetAssembly.GetType(className);
             if (definingType == null) return null;
 
-            var method = definingType.GetMethod(methodName);
+            var method = definingType.GetMethods().Where(o => o.Name == methodName).OrderBy(o => o.GetParameters().Length).FirstOrDefault();
```

Comment:
What happens in this code when the targetAssembly is compiled for .NET 2.0? My assumption has been that both the discovery and execution processes run under the targeted .NET version. If so, this would be a problem if executed in such a situation. Of course, we **shouldn't** ever get here in that case, but do we need some sort of guard clause somewhere to be safe?